src/truncate.jl

```diff
@@ -25,8 +25,11 @@ function truncated(d::UnivariateDistribution, l::T, u::T) where {T <: Real}
     T2 = promote_type(T, eltype(d))
     lcdf = isinf(l) ? zero(T2) : T2(cdf(d, l))
     ucdf = isinf(u) ? one(T2) : T2(cdf(d, u))
-    tp = ucdf - lcdf
-    Truncated(d, promote(l, u, lcdf, ucdf, tp, log(tp))...)
+    logcdf_l = isinf(l) ? T2(-Inf) : T2(logcdf(d, l))
+    logcdf_u = isinf(u) ? zero(T2) : T2(logcdf(d, u))
```

Comment:
The changes in https://github.com/JuliaStats/Distributions.jl/pull/1348 should allow us to just use
```suggestion
    logcdf_l = logcdf(d, l)
    logcdf_u = logcdf(d, u)
```
Note, however, that this does not handle discrete distributions correctly - for these one would have to subtract the pdf of the lower bound from the cdf. But since it is already incorrect in the current release, this could be addressed in a separate PR.

Reply:
Similarly, one could drop `isinf` in the computation of `lcdf` and `ucdf`, and remove `T2` completely.

Reply:
Thanks for the comment! I have pushed a new commit to incorporate the suggested changes.